cloud/ec2_util.go

```diff
@@ -88,8 +88,11 @@ func AztoRegion(az string) string {
 
 // ec2StatusToEvergreenStatus returns a "universal" status code based on EC2's
 // provider-specific status codes.
-func ec2StatusToEvergreenStatus(ec2Status types.InstanceStateName) CloudStatus {
-	switch ec2Status {
+func ec2StatusToEvergreenStatus(ec2State *types.InstanceState) CloudStatus {
```

Comment:
Since we only use the name from ec2State, let's instead just take in the name.

Reply:
This prevents a bug where one place was doing a nil check while the other wasn't. Which would cause a panic. I'm worried about reverting this, then more divergence between the functions happening